includes/class-wc-payments-account.php

```diff
@@ -108,7 +108,8 @@ public function init_hooks() {
 		add_action( 'admin_init', [ $this, 'maybe_redirect_after_plugin_activation' ], 11 ); // Run this after the WC setup wizard and onboarding redirection logic.
 		add_action( 'admin_init', [ $this, 'maybe_redirect_by_get_param' ], 12 ); // Run this after the redirect to onboarding logic.
 		add_action( 'admin_init', [ $this, 'maybe_redirect_from_settings_page' ] );
-		add_action( 'admin_init', [ $this, 'maybe_redirect_from_onboarding_page' ] );
+		add_action( 'admin_init', [ $this, 'maybe_redirect_from_onboarding_wizard_page' ] );
+		add_action( 'admin_init', [ $this, 'maybe_redirect_from_connect_page' ] );
```

Comment:
I am curious why having separate entrypoint rather than doing it in maybe_handle_onboarding?

Reply:
I mean similar is also happening [here](https://github.com/Automattic/woocommerce-payments/blob/00eb23816876c3c41935dcb8b8d40af7aa2b2e82/includes/class-wc-payments-account.php#L1062-L1062), but for fully onboarded accounts

Reply:
Like I explained in the big comment in `maybe_handle_onboarding`, we should not confuse connect links with connect _page_ links. `maybe_redirect_from_connect_page` handles possible redirects from the Connect page, not from connect links. I hope this makes sense.

Reply:
Yeah, gotcha. Maybe worth mentioning in very short in `maybe_redirect_from_connect_page` too (like @see ...), so you can reach it from two entrypoints 🙂

Reply:
[Added](https://github.com/Automattic/woocommerce-payments/pull/9105/commits/12e765046aa1b158aab7d62499cd6a7ea0705efe) docs.